### ISSUE TITLE
Is there a way to do complex path expansion in agensgraph?

### ISSUE THREAD

**Issue by jafri** [bitnine-oss/agensgraph#501]
In neo4j, you have APOC queries that enable variable length path traversals.

"The apoc.path.expand procedure makes it possible to do variable length path traversals where you can specify the direction of the relationship per relationship type and a list of Label names which act as a "whitelist" or a "blacklist" or define end nodes for the expansion. The procedure will return a list of Paths in a variable name called "path".

Source: http://neo4j-contrib.github.io/neo4j-apoc-procedures/3.5/path-finding/path-expander/

Is it possible to do a similar type of complex path expansion with agensgraph?

**Comment by joefagan** [comment]
Yes it is possible to similar traversals/queries but not exactly the same.

**Comment by jafri** [comment]
@joefagan are there any examples I could look at on how to achieve this without APOC? 

Its a major point that we are considering when evaluating whether to move from neo4j to agensgraph, would be great to see how similar functionality could be achieved.

**Comment by erik-overdahl** [comment]
This puts restrictions on the nodes in a graph path

```
MATCH path=(n:vlabel_a)-[* min_depth..max_depth]->()
WHERE *initial conditions on first node and last node*
WITH path, unnest(nodes(p)) AS n
WHERE NONE(x IN labels(n) WHERE x IN [*blacklisted labels*])
WITH p, collect(n) AS regroup
RETURN p;
```

add `ALL`, `ANY`, or `SINGLE`, or substitute them for `NONE`, to whitelist in various ways.

Not sure how to do the different directions for different relationship types.
